scripts/dts/gen_defines.py

```diff
@@ -111,6 +114,22 @@ def parse_args():
     return parser.parse_args()
 
 
+def relativize(path):
+    # If 'path' is within $ZEPHYR_BASE, returns it relative to $ZEPHYR_BASE,
+    # with a "$ZEPHYR_BASE/..." hint at the start of the string. Otherwise,
+    # returns 'path' unchanged.
```

Comment:
While the other fix is really under-commented, maybe this one is a bit over-commented?

Also why no pydoc syntax? (if anything)

Reply:
I usually do doc comments unless it's part of the API for some library. Another way of marking it internal, kinda. If it was part of a library (`gen_defines.py` is just a script), I'd put `_` before the name too.

Not saying that's the one true way to do it, but there's a method to the madness. ;)